sound/soc/sof/trace.c

```diff
@@ -144,6 +146,10 @@ int snd_sof_init_trace_ipc(struct snd_sof_dev *sdev)
 	params.buffer.size = sdev->dmatb.bytes;
 	params.buffer.pages = sdev->dma_trace_pages;
 	params.stream_tag = 0;
+	shift = jiffies_64 - sdev->bootup_timestamp;
+	/* jiffies to ms */
+	shift = shift * 1000 / HZ;
+	params.timestamp = shift;
```

Comment:
that also looks like a non backwards compatible change? we'd need to create two separate structures and play with v3/v4.
@lgirdwood I can't recall why we don't have reserved fields in that trace structure? Or maybe we need to pass the time in a different struct?

Reply:
@plbossart this change is definitely not backwards compatible. But is backwards compatibility a requirement or can we make it a new enhancement and bump up the major version to make it incompatible with older versions?

Reply:
@plbossart @ranj063 Can we create a new ipc command type to send timestamp? e.g. SOF_IPC_TRACE_TIMESTAMP. So FW will not receive the ipc if kernel version is old and FW will return -EINVAL and ignore the ipc if FW version is old. It seems to be backwards compatible if we do so.